src/finding/mod.rs

```diff
@@ -267,18 +268,54 @@ impl<'w> FindingBuilder<'w> {
     }
 
     pub(crate) fn build(self, workflow: &'w Workflow) -> Result<Finding<'w>> {
+        let locations = self
+            .locations
+            .iter()
+            .map(|l| l.clone().concretize(workflow))
+            .collect::<Result<Vec<_>>>()?;
+
+        let should_ignore = self.ignored_from_inlined_comment(workflow, &locations, self.ident);
+
         Ok(Finding {
             ident: self.ident,
             desc: self.desc,
             determinations: Determinations {
                 confidence: self.confidence,
                 severity: self.severity,
             },
-            locations: self
-                .locations
-                .into_iter()
-                .map(|l| l.concretize(workflow))
-                .collect::<Result<Vec<_>>>()?,
+            locations,
+            ignored: should_ignore,
         })
     }
+
+    fn ignored_from_inlined_comment(
+        &self,
+        workflow: &Workflow,
+        locations: &[Location],
+        id: &str,
+    ) -> bool {
+        let document_lines = &workflow.document.source().lines().collect::<Vec<_>>();
+        let line_ranges = locations
```

Comment:
Small thing: I'm wondering if it makes sense to scan all locations like this, or require that the ignore comment is present only on the first location in the finding.

I don't feel strongly either way, but my thinking is that forcing it to be only on the first location keeps it at the "top" in general, and reduces the amount of scanning we need to do. Thoughts?

Reply:
I gave a small try on the following change (local only)

```rust
        if let Some(first_location) = locations.first() {
            let start_line = first_location.concrete.location.start_point.row;
            let end_line = first_location.concrete.location.end_point.row + 1;
            let inlined_ignore = format!("# zizmor: ignore[{}]", id);

            for document_line in start_line..end_line {
                if let Some(line) = document_lines.get(document_line) {
                    if line.rfind(&inlined_ignore).is_some() {
                        return true;
                    }
                }
            }
        }
```

and tried against the following workflow, with exercises an `artipacked` finding

```yml
on:
  push:
    branches:
      - master
  workflow_dispatch:

jobs:
  artipacked:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Upload artifact
        uses: actions/upload-artifact@v4
        with:
          name: artipacked
          path: . # zizmor: ignore[artipacked]
```

I got

```
error[artipacked]: credential persistence through GitHub Actions artifacts
  --> tests/test-data/artipacked.yml:13:9
   |
13 |         - uses: actions/checkout@v4
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^ does not set persist-credentials: false
14 |
15 |         - name: Upload artifact
   |  _________^
16 | |         uses: actions/upload-artifact@v4
17 | |         with:
18 | |           name: artipacked
19 | |           path: . # zizmor: ignore[artipacked]
   | |______________________________________________^ may leak the credentials persisted above
   |
   = note: audit confidence → High
```

I have mixed feelings whether or not users may get confused with this, since we are not ignoring as requested. Strictly speaking from the user perspective, may take is going after for all `locations`. 

Thinking on future audits that may track two or more `locations` (right now only `artipacked` and `template-injection` does that from what I could learn), the behavior for inlined ignores stay the same if we scan all of them, so we don't need to care about special cases in which the meaningful `location` to check for an `ignore` is not the first one.

Wdyt?

Reply:
Oh yeah, that's a good point -- I could see people expecting that `ignore` to work.